network/p2p/connection/connManager_test.go

```diff
@@ -88,3 +93,71 @@ func generatePeerInfo(t *testing.T) peer.ID {
 	require.NoError(t, err)
 	return pInfo.ID
 }
+
+// TestConnectionManager_Watermarking tests that the connection manager prunes connections when the number of connections
+// exceeds the high watermark and that it does not prune connections when the number of connections is below the low watermark.
+func TestConnectionManager_Watermarking(t *testing.T) {
+	sporkId := unittest.IdentifierFixture()
+	ctx, cancel := context.WithCancel(context.Background())
+	signalerCtx := irrecoverable.NewMockSignalerContext(t, ctx)
+	defer cancel()
+
+	cfg := &connection.ManagerConfig{
+		HighWatermark: 4,               // whenever the number of connections exceeds 4, connection manager prune connections.
+		LowWatermark:  2,               // connection manager prune connections until the number of connections is 2.
+		GracePeriod:   1 * time.Second, // extra connections will be pruned if they are older than a second (just for testing).
+		SilencePeriod: 5 * time.Second, // connection manager prune checking kicks in every 5 seconds (just for testing).
```

Comment:
```suggestion
		GracePeriod:   500 * time.Millisecond, // extra connections will be pruned if they are older than a second (just for testing).
		SilencePeriod:  time.Second, // connection manager prune checking kicks in every 5 seconds (just for testing).
```
We can reduce the total test run time to 2s from 10s , I've tested this there is no flakiness.